File: python3/evaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        
        if len(tokens) == 1:
            return int(tokens[0])
        
        num_stack = []
        
        for i in tokens:
            if i == "+" or i == "-" or i == "*" or i == "/":
                num = int(num_stack.pop())
                if i == "+":
                    ans = int(num_stack.pop()) + num
                if i == "-":
                    ans = int(num_stack.pop()) - num
                if i == "*":
                    ans = int(num_stack.pop()) * num
                if i == "/":
                    ans = int(num_stack.pop()) / num
                num_stack.append(ans)
            else:
                num_stack.append(i)

        return int(ans)

File: golang/evaluateReversePolishNotation.go
func evalRPN(tokens []string) int {
	var tokenStack []int
	var num int
	var res int
	for _, token := range tokens {
		if token == "+" || token == "-" || token == "*" || token == "/" {
			switch token {
			case "+":
				res = tokenStack[len(tokenStack)-2] + tokenStack[len(tokenStack)-1]

			case "-":
				res = tokenStack[len(tokenStack)-2] - tokenStack[len(tokenStack)-1]

			case "*":
				res = tokenStack[len(tokenStack)-2] * tokenStack[len(tokenStack)-1]

			case "/":
				res = tokenStack[len(tokenStack)-2] / tokenStack[len(tokenStack)-1]

			}
			tokenStack = tokenStack[:len(tokenStack)-2]
			tokenStack = append(tokenStack, res)
		} else {
			num, _ = strconv.Atoi(token)
			tokenStack = append(tokenStack, num)
		}
	}
	return tokenStack[0]
}
